kie-soup-dataset/kie-soup-dataset-api/src/main/java/org/dashbuilder/dataset/def/PrometheusDataSetDef.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.dashbuilder.dataset.def;
+
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+
+import org.dashbuilder.dataprovider.DataSetProviderType;
+import org.dashbuilder.dataset.validation.groups.PrometheusDataSetDefValidation;
+
+public class PrometheusDataSetDef extends DataSetDef {
+
+    @Size(min = 7, groups = {PrometheusDataSetDefValidation.class})
+    @NotNull(groups = {PrometheusDataSetDefValidation.class})
+    protected String serverUrl;
+
+    @Size(min = 1, groups = {PrometheusDataSetDefValidation.class})
+    @NotNull(groups = {PrometheusDataSetDefValidation.class})
+    protected String query;
+
+    private String user;
+    private String password;
+
+    public PrometheusDataSetDef() {
+        super.setProvider(DataSetProviderType.PROMETHEUS);
+        serverUrl = "http://localhost:9090";
```

Comment:
Should this port be a fixed value?

Reply:
This value should be overriden later by actual values - I set the default as the default Prometheus port, but usually it is overriden